packages/assets-controllers/src/TokenDetectionController.test.ts

```diff
@@ -523,73 +489,44 @@ describe('TokenDetectionController', () => {
     const onPreferencesStateChange = sinon.stub().callsFake((listener) => {
       preferencesStateChangeListener = listener;
     });
-    tokenDetection = new TokenDetectionController(
-      {
-        onPreferencesStateChange,
-        onTokenListStateChange: stub,
-        onNetworkStateChange: stub,
-        getBalancesInSingleCall: getBalancesInSingleCallMock,
-        addDetectedTokens: stub,
-        getTokensState: () => tokensController.state,
-        getTokenListState: () => tokenList.state,
-        getNetworkState: () => defaultNetworkState,
-        getPreferencesState: () => preferences.state,
-        getNetworkClientById: jest.fn(),
-      },
-      {
-        disabled: false,
-        isDetectionEnabledForNetwork: true,
-        isDetectionEnabledFromPreferences: false,
-        selectedAddress: '0x1',
-      },
-    );
+    tokenDetection = new TokenDetectionController({
+      chainId: ChainId.mainnet,
+      selectedAddress: '0x1',
+      onPreferencesStateChange,
+      getBalancesInSingleCall: getBalancesInSingleCallMock,
+      addDetectedTokens: stub,
+      getTokensState: () => tokensController.state,
+      getPreferencesState: () => preferences.state,
+      messenger: buildTokenDetectionControllerMessenger(controllerMessenger),
+    });
+    await tokenDetection.start();
```

Comment:
The reason I was asking why we needed to call `start` now is because the preferences changing should cause state updates by itself, so this controller shouldn't need to be started in order for the test to pass. Did you find that not to be true?

Reply:
Removing the `start()` calls here break both tests. Trying to find a fix.

Reply:
Fixed so that `selectedAddress` is changed in the updated preferences: https://github.com/MetaMask/core/pull/3609/commits/5d602a3b337a181bcac59b3160457464ee1e7820

Not sure why the test was passing originally when the `selectedAddress` wasn't being modified, but it looks like it's exhibiting correct behavior now.